File: frontend/src/utils/phone.js
/** Cambodia local phone: 0 + 8–9 digits (9–10 total), e.g. 0978843978 */

const CAMBODIA_PHONE_RE = /^0[1-9]\d{7,8}$/

export function normalizeCambodiaPhone(value) {
  let digits = String(value || '').replace(/\D/g, '')
  if (digits.startsWith('855')) {
    digits = `0${digits.slice(3)}`
  }
  return digits.slice(0, 10)
}

export function isValidCambodiaPhone(value) {
  return CAMBODIA_PHONE_RE.test(normalizeCambodiaPhone(value))
}

File: backend/utils/phone.py
import re

from rest_framework import serializers

CAMBODIA_PHONE_RE = re.compile(r'^0[1-9]\d{7,8}$')
CAMBODIA_PHONE_ERROR = 'Enter a Cambodia phone number starting with 0 (9–10 digits).'


def normalize_cambodia_phone(value: str) -> str:
    digits = re.sub(r'\D', '', str(value or ''))
    if digits.startswith('855'):
        digits = f'0{digits[3:]}'
    return digits[:10]


def validate_cambodia_phone(value: str, allow_blank: bool = False) -> str:
    raw = str(value or '').strip()
    if not raw:
        if allow_blank:
            return ''
        raise serializers.ValidationError(CAMBODIA_PHONE_ERROR)

    normalized = normalize_cambodia_phone(raw)
    if not CAMBODIA_PHONE_RE.fullmatch(normalized):
        raise serializers.ValidationError(CAMBODIA_PHONE_ERROR)
    return normalized
